pkg/plugin/keyutil/pem_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keyutil
+
+import (
+	"testing"
+
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestReadRSAPrivateKey(t *testing.T) {
+	t.Parallel()
+
+	testPrivateRSAKeyString, testPrivateRSAKey := TestGenerateRsaPrivateKey(t)
+
+	cases := []struct {
+		name        string
+		pkPEMString string
+		wantErr     string
+	}{
+		{
+			name:        "success",
+			pkPEMString: testPrivateRSAKeyString,
+		},
+		{
+			name:        "invalid_pem",
+			pkPEMString: "invalid_format",
+			wantErr:     "failed to decode PEM formated key",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			gotPK, err := ReadRSAPrivateKey(tc.pkPEMString)
+			if err != nil {
+				if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+					t.Errorf(diff)
+				}
+			} else {
+				if diff := cmp.Diff(gotPK, testPrivateRSAKey); diff != "" {
+					t.Errorf("rsa private key got unexpected diff (-want,+got):\n%s", diff)
+				}
+			}
+		})
```

Comment:
Is this the correct logic? Wouldn't we always want to compare the error and the expected output?

Reply:
updated.